pkg/asset/machines/master.go

```diff
@@ -253,6 +314,13 @@ func (m *Master) Files() []*asset.File {
 		files = append(files, m.UserDataFile)
 	}
 	files = append(files, m.MachineConfigFiles...)
+	// Hosts refer to secrets, so place the secrets before the hosts
+	// to avoid unnecessary reconciliation errors.
```

Comment:
Ordering here doesn't matter, does it?  These will be pushed in the order based on their filenames.

Reply:
Yeah, I think that comment was from before I fully understood how that part worked and I've just been carrying it along.